include/net/lwm2m_client_utils_location.h

```diff
@@ -32,30 +30,161 @@
 #define LOCATION_ASSIST_NEEDS_INTEGRITY			BIT(8)
 
 #if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS)
-struct gnss_agps_request_event {
-	struct app_event_header header;
+/**
+ * @brief Set the A-GPS request mask
+ *
+ * @param agps_req The A-GPS request coming from the GNSS module.
+ */
+void location_assistance_agps_set_mask(const struct nrf_modem_gnss_agps_data_frame *agps_req);
 
-	struct nrf_modem_gnss_agps_data_frame agps_req;
-};
+/**
+ * @brief Send the A-GPS assistance request to LwM2M server
+ *
+ * @param ctx LwM2M client context for sending the data.
+ * @param confirmable Boolean value to indicate should the server return confirmation.
+ * @return Returns a negative error code (errno.h) indicating
+ *         reason of failure or 0 for success.
+ */
+int location_assistance_agps_request_send(struct lwm2m_ctx *ctx, bool confirmable);
+#endif
 
-APP_EVENT_TYPE_DECLARE(gnss_agps_request_event);
+/**
+ * @brief Send the Ground Fix request to LwM2M server
+ *
+ * @param ctx LwM2M client context for sending the data.
+ * @param confirmable Boolean value to indicate should the server return confirmation.
+ * @return Returns a negative error code (errno.h) indicating
+ *         reason of failure or 0 for success.
+ */
+int location_assistance_ground_fix_request_send(struct lwm2m_ctx *ctx, bool confirmable);
 
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_PGPS)
```

Comment:
```suggestion
```

I would propose that we declare all APIs in the public header without any #ifdeffing.

This would allow application to use syntax like
```
void do_pgps_reg()
{
    if (IS_ENABLED(CONFIG_APP_USE_PGPS)) {
        location_assistance_pgps_request_send();
    }
}
```

Instead of
```
void do_pgps_reg()
{
#if defined(CONFIG_APP_USE_PGPS) 
    location_assistance_pgps_request_send();
#endif
}
```

Reply:
Just chiming in that I also prefer this pattern, and kind of feel we should adopt it as widely as possible throughout NCS

Plus, @jayteemo has recently discovered that ifdef'd header declarations don't get added to docs, which further supports this approach

Reply:
re: docs for #if blocks
https://github.com/nrfconnect/sdk-nrf/pull/9068
it looks like this file needs to be manually updated if there are new #if blocks added.